packages/xchain-mayachain-amm/src/mayachain-action.ts

```diff
@@ -87,47 +93,71 @@ export class MayachainAction {
       }
     }
 
-    // EVM actions
+    // EVM actions and Radix go through the router
     const mayachainQuery: MayachainQuery = new MayachainQuery()
 
     const inboundDetails = await mayachainQuery.getChainInboundDetails(assetAmount.asset.chain)
     if (!inboundDetails.router) throw Error(`Unknown router for ${assetAmount.asset.chain} chain`)
 
-    const isERC20 = isProtocolERC20Asset(assetAmount.asset)
+    if (isRadixChain(assetAmount.asset.chain)) {
+      // Radix
+      const hash = await wallet.transfer({
+        asset: assetAmount.asset,
+        recipient: inboundDetails.router,
+        amount: assetAmount.baseAmount,
+        methodToCall: {
+          address: inboundDetails.router,
+          methodName: 'user_deposit',
+          params: [
+            generateAddressParam(await wallet.getAddress(AssetXRD.chain)),
+            generateAddressParam(recipient),
+            generateBucketParam(0),
+            generateStringParam(memo),
+          ],
+        },
+      })
+      return {
```

Comment:
Do you need the next `else` if you have this `return` here?

Reply:
Updated 💪🏻